src/context/AppContext.tsx

```diff
@@ -47,34 +49,37 @@ export const AppContextProvider = ({ children }: { children: ReactNode }) => {
   const [userLoading, setUserLoading] = useState(false);
   const [walletConnecting, setWalletConnecting] = useState(false);
 
-  const updateUser = useCallback((address?: string) => {
-    if (address) {
-      try {
-        setUserLoading(true);
-        const apolloClient = getContext(ContextModule.ApolloClient);
-        const query = apolloClient.query<
-          GetCurrentUserQuery,
-          GetCurrentUserQueryVariables
-        >({
-          query: GetCurrentUserDocument,
-          variables: { address },
-          fetchPolicy: 'network-only',
-        });
-        query.then(({ data }) => {
+  const updateUser = useCallback(
+    async (address?: string, shouldBackgroundUpdate = false) => {
```

Comment:
Per the comment below, now that `userLoading` is being set to true as expected, it's useful that we have some way of performing context updates in the "background", i.e. without prompting the display of a loading spinner. E.g when updating user settings or uploading a user avatar.